app/Http/Controllers/ProductController.php

```diff
@@ -13,36 +14,42 @@ class ProductController extends Controller
     public function index()
     {
         $products = Product::select('id', 'name', 'brand_id', 'color_id', 'size_id', 'status')->paginate(10);
-          
+
         return view('admin.pages.product.index', compact('products'));
     }
     public function add()
     {
         $brands = Brand::select('id', 'name')->get();
         $colors = Color::select('id', 'name')->get();
-        $sizes = Size::select('id', 'name')->get();    
-        return view('admin.pages.product.form', compact('brands', 'colors', 'sizes'));
+        $sizes = Size::select('id', 'name')->get();
+        $categories = Category::select('id', 'name')->get();
+
+        return view('admin.pages.product.form', compact('brands', 'colors', 'sizes', 'categories'));
     }
     public function store(ProductValidation $request)
     {
-        Product::create($request->validated());
+        $product = Product::create($request->validated());
+        $product->attachCategories($request->input('category_ids'));
```

Comment:
`attachCategories` method move to here(controller) and it should be private function.